gravitee-am-service/src/main/java/io/gravitee/am/service/impl/PasswordPolicyServiceImpl.java

```diff
@@ -160,9 +172,30 @@ public Maybe<PasswordPolicy> retrievePasswordPolicy(io.gravitee.am.model.User us
     }
 
     private Maybe<PasswordPolicy> defaultPasswordPolicy(io.gravitee.am.model.User user) {
-        return passwordPolicyRepository.findByReference(user.getReferenceType(), user.getReferenceId())
-                // TODO replace this sort by a filter on default attribute during AM-2893
-                .sorted(Comparator.comparing(PasswordPolicy::getCreatedAt))
-                .firstElement();
+        return passwordPolicyRepository.findByDefaultPolicy(user.getReferenceType(), user.getReferenceId());
+    }
+
+    private Single<PasswordPolicy> setNewDefaultPolicy(ReferenceType referenceType, String referenceId, String policyId, User principal) {
+        return passwordPolicyRepository.findByReferenceAndId(referenceType, referenceId, policyId)
+                .switchIfEmpty(Single.error(() -> new PasswordPolicyNotFoundException(policyId)))
```

Comment:
TODO: I would do this control either in the JAX-RS resource or as a very first action in `setDefaultPasswordPolicy` to avoid updating the previous default policy if the new one doesn't exist.